hw/ip/aes/doc/_index.md

```diff
@@ -264,19 +271,65 @@ For block operation, software must initialize the AES unit as described in the p
 3. Wait for the INPUT_READY bit in {{< regref "STATUS" >}} to become `1`, i.e. wait for the AES unit to load Input Data Block `0` into the internal state register and start operation.
 4. Write Input Data Block `1` to the Input Data registers.
 
-Then for every Data Block `I=0,..,N-2`, software must:
+Then for every Data Block `I=0,..,N-3`, software must:
 1. Wait for the OUTPUT_VALID bit in {{< regref "STATUS" >}} to become `1`, i.e., wait for the AES unit to finish encryption/decryption of Block `I`.
    The AES unit then directly starts processing the previously input block `I+1`
 2. Read Output Data Block `I` from the Output Data register.
 3. Write Input Data Block `I+2` into the Input Data register.
    There is no need to check INPUT_READY as the cycle following OUTPUT_VALID being set the current input is loaded in.
 
-Once all blocks have been input the final data blocks `I=N-1,N` must be read out
+Once all blocks have been input the final data blocks `I=N-2,N-1` must be read out
 1. Wait for the OUTPUT_VALID bit in {{< regref "STATUS" >}} to become `1`, i.e., wait for the AES unit to finish encryption/decryption of Block `I`.
 2. Read Output Data Block `I` from the Output Data register.
 
 Note that interrupts are not provided, the latency of the AES unit is such that they are of little utility.
 
+The code snippet below shows how to perform block operation.
+
+```c
+  // Enable autostart, disable overwriting of previous output data
+  REG32(AES_CTRL(0)) =
+      mode << AES_CTRL_MODE |
+      (key_len & AES_CTRL_KEY_LEN_MASK) << AES_CTRL_KEY_LEN_OFFSET |
+      0x0 << AES_CTRL_MANUAL_START_TRIGGER |
+      0x0 << AES_CTRL_FORCE_DATA_OVERWRITE;
+
+  // Write Input Data Block 0
+  for (int j = 0; j < 4; j++) {
+    REG32(AES_DATA_IN0(0) + j * 4) = input_data[j];
```

Comment:
I don't recall if we ever make any explicit mention of Endianness, but given that is a key point in the AES spec, it might be worth mentioning our Little Endianness.

Reply:
Thanks for your feedback @sjgitty , I will add this info.